framework/include/restart/DataIO.h

```diff
@@ -123,6 +123,13 @@ template<typename T>
 inline void
 dataStore(std::ostream & stream, T & v, void * /*context*/)
 {
+  #ifdef LIBMESH_HAVE_CXX11
```

Comment:
`is_polymorphic` is from `<type_traits>`, so change this check to 
```
#ifdef LIBMESH_HAVE_CXX11_TYPE_TRAITS
```
as that's more targeted to what you are actually doing.